kratos/solving_strategies/strategies/line_search_strategy.h

```diff
@@ -274,56 +341,93 @@ class LineSearchStrategy
         typename TSchemeType::Pointer pScheme = this->GetScheme();
         typename TBuilderAndSolverType::Pointer pBuilderAndSolver = this->GetBuilderAndSolver();
 
-        TSystemVectorType aux(b.size()); //TODO: do it by using the space
-        TSparseSpace::Assign(aux,0.5, Dx);
-
-        //compute residual without update
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double ro = TSparseSpace::TwoNorm(b);
-
-        //compute half step residual
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
-        TSparseSpace::SetToZero(b);
-        pBuilderAndSolver->BuildRHS(pScheme, BaseType::GetModelPart(), b );
-        double rh = TSparseSpace::TwoNorm(b);
-
-        //compute full step residual (add another half Dx to the previous half)
-        BaseType::UpdateDatabase(A,aux,b,MoveMesh);
+        TSystemVectorType aux(TSparseSpace::Size(b));
+        
+        double x1 = mFirstAlphaValue;
+        double x2 = mSecondAlphaValue;
+
+        bool converged = false;
+        int it = 0;
+        double xprevious = 0.0;
+
+        //Compute residual with 1 coefficient update (x1)
+        //since no update was performed yet, this includes an increment wrt the previous
+        //solution of x1*Dx
+        TSparseSpace::Assign(aux,x1-xprevious, Dx);
```

Comment:
you should not do x1 - xprevious ... otherwise it will not work in trilinos (the - operator). 
this needs to be done all by space functions

Reply:
But x1 and xprevious are doubles, that should work right?